### ISSUE TITLE
error when using math along with a variable name

### ISSUE THREAD

**Issue by chrisdrackett** [sasstools/sass-lint#809]
What version of Sass Lint are you using? `1.7.0`

Please include any relevant parts of your configuration

`space-around-operator: 0`
`  variable-name-format:
    - 1
    - convention: snakecase`

What did you do? Please include the actual source code causing the issue.

`$global_nav_width: $grid*10`

What did you expect to happen?

I expect this to output without errors.

What actually happened? Please include any error messages given to you by Sass Lint.

I get an error `Variable 'grid*10' should be written in snake_case.`

If you're using a IDE plugin have you tried the CLI too?

nope

**Comment by chrisdrackett** [comment]
looks like `$grid/4`, `$grid+4` work, in addition to `*`, `$grid-4` does not work

**Comment by BPScott** [comment]
The `$grid-4` case is a funky one. Sass allows hyphens in variable names, which means that fiddling with spacing around operators leads to different results (and potentially compilation errors):

```scss
$foo: 100;
$foo-20: 10;

$w: $foo - 20; // Does some maths - so $w == 100 -20 == 80
$x: $foo-20; // Uses the variable - so $x == 20
$y: $foo - 40; // Does some maths - so $ == 100 - 40 == 60
$z: $foo-40; // Explosions - Compilation error as it tries to use an unknown variable
```

Sassmeister: http://www.sassmeister.com/gist/312e4069430d4be672cd5202de4ef205

---

Thus it is impossible to enforce no space around operators if you want to do subtraction.

Sass-lint should probably be updated handle `*` in the same way as `+` and `/`, but it cant do that for hypens for minusing, because in the language`$foo-20` tries to use a variable called that, not take 20 away from `$foo` - and thus it falls afoul of your variable name formatting as you're trying to reference a variable that isn't snake_case.

**Comment by DanPurdy** [comment]
Thanks for bringing this up, I'll look into it. Also as @BPScott mentioned the minus in variable names like that is quite a well known quirk of the language which isn't something I'd expect to see ever resolved.
